core/storage/base_model/gae_models.py

```diff
@@ -1131,9 +1144,92 @@ def delete( # type: ignore[override]
             BaseModel.update_timestamps_multi(models_to_put_values)
             BaseModel.put_multi(models_to_put_values)
 
+    def get_datastore_keys_for_delete(
+        self
+    ) -> List[datastore_services.Key]:
+        """Get this model instance's Datastore keys.
```

Comment:
Explain this more -- what determines whether a key is in scope to be returned vs not?